src/heaps/fibonacci_heap.jl

```diff
@@ -0,0 +1,367 @@
+end
+
+# Get list of nodes attached to head (of a circular doubly linked list)
+function aslist(head::MutableFibonacciHeapNode)
+    nodelist, node, stop = MutableFibonacciHeapNode[], head, head
+    flag = false
+    while true
+        if node == stop
+            flag && break
+            flag = true
+        end
+        push!(nodelist, node)
+        node = node.right
+    end
+    return nodelist
+end
+
+function Base.show(io::IO, h::MutableFibonacciHeap{T}) where T
+    if h.nodeCount == 0
+        print(io,"FibonacciHeap{$T}()")
+        return
+    end
+    print(io, "FibonacciHeap([")
+    print(io,"$(h.rootList.value)")
+    show_heap_helper(io, h.rootList,h)
+    print(io,"])")
+end
+
+
+function show_heap_helper(io::IO,rootnode::MutableFibonacciHeapNode{T},h::MutableFibonacciHeap{T}) where T
+    node = rootnode
+    while node !== nothing
+        if node !== h.rootList
+            print(io,",$(node.value)")
+        end
+        if node.child !== nothing
+            show_heap_helper(io, node.child,h)
+        end
+        if node.right == rootnode
+            break
+        end
+        node = node.right
+    end
+end
+
+function findNode(head::MutableFibonacciHeapNode{T}, key::T) where T
+    found::Union{MutableFibonacciHeapNode,Nothing} = nothing
+    temp = head
+    temp.flag = true
+    if temp.value === key
+        found = temp; 
+        temp.flag = false; 
+        return found; 
+    end
+    if found === nothing
+        if temp.child !== nothing
+            return findNode(temp.child, key); 
+        end
+        if (temp.right).flag === false
+            return findNode(temp.right, key); 
+        end
+    end
+    temp.flag = false;
+
+end
+#################################################
+#
+#   interfaces
+#
+#################################################
+
+length(h::MutableFibonacciHeap) = h.nodeCount
+
+isempty(h::MutableFibonacciHeap) = h.nodeCount == 0
+
+# function sizehint!(h::MutableFibonacciHeap, s::Integer)
+#     sizehint!(h.nodes, s)
+#     sizehint!(h.node_map, s)
+#     return h
+# end
+
+"""
+        top(h::MutableFibonacciHeap)
+
+Returns the value of top element in heap `h`
+"""
+@inline top(h::MutableFibonacciHeap) = h.minNode.value
+
+"""
+        top_with_handle(h::MutableFibonacciHeap)
+
+Returns the top element of the heap `h` and its handle.
+"""
+@inline minimum(h::MutableFibonacciHeap) = h.minNode.value
+
+@inline top_with_handle(h::MutableFibonacciHeap) = h.minNode
+
+@inline minimum_with_handle(h::MutableFibonacciHeap) = h.minNode
+"""
+    push!{T}(h::MutableFibonacciHeap{T},data::T)
+
+Insert element into the heap `h` and return its handle.
+"""
+function push!(root::MutableFibonacciHeap,data::T) where T
+    node = MutableFibonacciHeapNode(data)
+    node.left = node.right = node
+    _merge_with_root_list(root, node)
+    if root.minNode === nothing || node.value < root.minNode.value
+        root.minNode = node
+    end
+    root.nodeCount += 1
+    return node
+end
```

Comment:
The signature of this method both in the code and the documentation is a bit  off:
* First, the docs write `pop!{T}`, even though functions are not parameterized.
* Second, `MutableFibonacciHeap` in the type signature should probably be `MutableFibonacciHeap{T}`? I think it's preferrable instead to specify the type of `data` and instead have `dataT = convert(T, data)` in the first line. This will allow e.g. pushing a `UInt` to a `MutableFibonacciHeap{Int}`.